app/src/org/commcare/activities/CommCareSetupActivity.java

```diff
@@ -508,6 +509,11 @@ protected void deliverError(CommCareSetupActivity receiver,
         }
     }
 
+    private int determineAuthorityForInstall() {
+        return lastInstallMode == INSTALL_MODE_OFFLINE ?

```

Comment:
Hm, strictly speaking the authority should be more about the reference connector than the specific method here. You can use the URL install or barcode install to direct to a local file, for instance.

I don't think it's important enough to drag everything out now, but would be worth at least documenting that limitation here as a cue for someone doing a bugfix in the future.